metabase/metabase
# Refactor question ViewHeader component

This PR is the first step for updating the question page for datasets (feature epic: #18659). ViewHeader is displayed for any question (saved vs ad-hoc, native vs GUI, etc.), so it has lots of possible states. This PR adds unit tests for the component and refactors it to simplify subsequent changes.

* turned into a functional component
* some visual components are extracted as styled-components
* the component itself is broken down into left side (are different for saved vs ad-hoc question) and right side components
* reordered imports

### To Verify

1. For saved GUI, ad-hoc GUI, saved native and ad-hoc native questions
2. Open the question page
3. Make sure there are no changes in the header's look and behavior

We have some Percy coverage for questions already. I've also submitted a PR with table visual tests that also cover ad-hoc header view #18826
